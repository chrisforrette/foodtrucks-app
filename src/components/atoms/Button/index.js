import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.blue};
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: 1rem;
  line-height: 1em;
  outline: none;
  padding: 0.6rem 1em;
  text-decoration: none;
  transition: all 200ms ease-in;

  &:hover:enabled {
    background-color: #fff;
    color: ${({ theme }) => theme.colors.blue};
  }

  &:disabled {
    opacity: 0.4;
  }

  &.hide {
    opacity: 0;
  }
`

export default ({ hide, children, ...props }) => <Button className={ hide ? 'hide' : '' } {...props}>{children}</Button>
