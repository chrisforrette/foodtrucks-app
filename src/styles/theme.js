// Breakpoints

export const breakpoints = {
  small: '600px',
  medium: '900px',
  large: '1200px',
  xlarge: '1800px'
}

// Fonts

// const fontHelvetica = '"HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif'
const fontHelveticaLight = '"HelveticaNeue-Light", "Helvetica Neue Light", "Helvetica Neue", Helvetica, Arial, sans-serif'
const fontHelveticaBold = '"HelveticaNeue-Bold", "Helvetica Neue Bold", "Helvetica Neue", Helvetica, Arial, sans-serif'

export const fonts = {
  heading: {
    primary: fontHelveticaBold
  },
  body: {
    primary: fontHelveticaLight
  }
}

// Colors

export const colors = {
  // Grey
  darkGrey: '#333',
  grey: '#999',
  lightGrey: '#eee'
}

export default {
  breakpoints,
  fonts,
  colors
}
