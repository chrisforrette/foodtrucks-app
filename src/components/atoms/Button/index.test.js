import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './index'
import theme from '../../../styles/theme'

it('should match snapshot', () => {
  const rendered = mount(<Button
    theme={theme}
    hide={false}
    mapBoundsChanged={false}>Click Me</Button>
  )
  expect(toJson(rendered)).toMatchSnapshot()
})
