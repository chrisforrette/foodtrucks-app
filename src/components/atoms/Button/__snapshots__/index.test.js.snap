// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  background-color: #41a6f4;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-size: 1rem;
  line-height: 1em;
  outline: none;
  padding: 0.6rem 1em;
  text-decoration: none;
  -webkit-transition: all 200ms ease-in;
  transition: all 200ms ease-in;
}

.c0:hover:enabled {
  background-color: #fff;
  color: #41a6f4;
}

.c0:disabled {
  opacity: 0.4;
}

.c0.hide {
  opacity: 0;
}

<Component
  hide={false}
  mapBoundsChanged={false}
  theme={
    Object {
      "breakpoints": Object {
        "large": "1200px",
        "medium": "900px",
        "small": "600px",
        "xlarge": "1800px",
      },
      "colors": Object {
        "blue": "#41a6f4",
        "darkGrey": "#333",
        "grey": "#999",
        "lightGrey": "#eee",
      },
      "fonts": Object {
        "body": "\\"HelveticaNeue-Light\\", \\"Helvetica Neue Light\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "heading": "\\"HelveticaNeue-Bold\\", \\"Helvetica Neue Bold\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
      },
    }
  }
>
  <styled.button
    className=""
    mapBoundsChanged={false}
    theme={
      Object {
        "breakpoints": Object {
          "large": "1200px",
          "medium": "900px",
          "small": "600px",
          "xlarge": "1800px",
        },
        "colors": Object {
          "blue": "#41a6f4",
          "darkGrey": "#333",
          "grey": "#999",
          "lightGrey": "#eee",
        },
        "fonts": Object {
          "body": "\\"HelveticaNeue-Light\\", \\"Helvetica Neue Light\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "heading": "\\"HelveticaNeue-Bold\\", \\"Helvetica Neue Bold\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        },
      }
    }
  >
    <button
      className="c0"
    >
      Click Me
    </button>
  </styled.button>
</Component>
`;
